### DISCUSSION TITLE
How to implement PopUntil?

### DISCUSSION THREAD

**Discussion by bw-flagship** [csells/go_router#326] [Q&A] [answered]
Thanks for the remarkable package!
Sometimes I want to pop all pages until I arrivte at my home page.

I found this way to implement it:

```
  void popUntil(bool Function(String) predicate) {
    while (!predicate(_router.routerDelegate.location)) {
      _router.pop();
    }
  }

popUntil((location) => !location.contains("home"));
```

This is not perfect because the location contains all the parameters, so I have to use the contains-operator.
Is there any better way to archive this?

Thanks!

**Comment by csells** [comment, marked as answer]
You don't need to do that. A simple `context.go('/')` will replace the entire stack of pages with just your homepage.
